### ISSUE TITLE
Simplest model to try narrowing RF to evaluate?

### ISSUE THREAD

**Issue by daisukelab** [kkoutini/cpjku_dcase20#1]
Hello, thank you for making your code open to the public.

I'd like to try to train CP-ResNet with my datasets, then trying to figure out which model is the best to try.
Simply to say, I'm wondering which is the simplest for trying just one extra parameter rho from the original ResNet.
My objective is to change the RF of ResNet and see what happens.

I appreciate if any recommendation could be made.

Thanks in advance :)

P.S. I also have to find your code for converting audio to spectrogram as input to the model.
P.S.2 I'd like to evaluate the pretrained model in various downstream tasks like speech commands, do you also have any recommendations from your pretrained weights?

**Comment by kkoutini** [comment]
Hi, thank you for your interest!
If I understood correctly, you'd like to try different rho values (which corresponds to different max RF sizes TABLE II in [1]) 
then I think most of the architectures, would behave similar to each other.  I would try the frequency-damped variant `cp_resnet_freq_damp.py` since it has a larger optimal RF range (Figures 9 and 12, 13 in [1] although in fig 13. it reduces the best performance).  
To use any architecture, you can import the function  `get_model_based_on_rho`  from any file in [models](https://github.com/kkoutini/cpjku_dcase20/blob/master/models/). 
Maybe also the Colab notebook is helpful, showing a simple setup  [here](https://colab.research.google.com/github/kkoutini/cpjku_dcase20/blob/master/cpjku_dcase20_colab.ipynb#scrollTo=LlRzKCm8-zWf).


You can find the spectrogram processior with librosa here:
[Audio processors](https://github.com/kkoutini/cpjku_dcase20/blob/master/audioprocessors.py). but I believe any audio processor should show similar RF/Accuracy curves as long as you have the same number of mel-bins: 256. ( in the tasks I checked  the RF over time is less important than over Frequency see  Fig. 7. in [1] )
 
Regarding the transfer learning from the pre-trained weights: I have no experience with this. but I'm curious about this :)

I hope this helps, and let me know, if something isn't clear!

[1] https://arxiv.org/pdf/2105.12395.pdf

**Comment by daisukelab** [comment]
Hi @kkoutini, thank you so much for your kind help.

I think I could instantiate a model object.

```python
>>> from models import cp_resnet_freq_damp

>>> model = cp_resnet_freq_damp.get_model_based_on_rho(rho=0, arch="cp_resnet_freq_damp",
                                               model_config_overrides={
                                                   'input_shape': [10, 1, -1, -1],
                                                   'stop_before_global_avg_pooling': True}).to('cuda')
>>> model(torch.rand((1, 1, 64, 96)).to('cuda')).shape

156 0.041666666666666664 0.07715167498104596
157 0.125 0.07715167498104596
158 0.125 0.07715167498104596
159 0.125 0.07715167498104596
160 0.08838834764831845 0.07715167498104596
161 0.08838834764831845 0.07715167498104596
162 0.08838834764831845 0.07715167498104596
163 0.08838834764831845 0.07715167498104596
164 0.0625 0.07715167498104596
165 0.0625 0.07715167498104596
166 0.0625 0.07715167498104596
167 0.0625 0.07715167498104596
torch.Size([1, 10, 3, 5])
```

And also thanks to the link to the [Audio processors](https://github.com/kkoutini/cpjku_dcase20/blob/master/audioprocessors.py), I could figure out what are the original FFT parameters (oh it's 22,050 Hz).

Plus, I could try changing rho and confirmed that the field of view varies, it's really nice!

Then I will try your model in my study.
I hope I can cite your paper eventually in the near future.

Thank you again, I appreciate for making your code open and kindly advice.
